### PR TITLE
Feature/dashboard nfts

### PR DESCRIPTION
**Pull request by mmahdigh**

### CODE REVIEW

File: src/app/dashboard/utils/nft-balance-opensea.ts
Comment (mmahdigh) [inline src/app/dashboard/utils/nft-balance-opensea.ts:1]: OpenSea APIs are very convenient and greatly facilitate fetching NFTs and their corresponding stats. (e.g. floor price)

However, their API is rate-limited and although they don't disclose the rate limit, it's predicted to be around a couple of requests per second. The bigger problem is they don't seem to provide not-rate-limited access to their APIs openly. So there's no pricing page or anything. They only have a [Google form](https://docs.google.com/forms/d/e/1FAIpQLScGpM0TibSpwoDM8_fx5lGwr4rqFDQh5wFFBEEuT-O-y-Vflg/viewform) to request a rate limit increase.

File: src/app/dashboard/data/top100-nfts.ts
[the remark this answers is not in the document]
Comment (jainkrati) [reply src/app/dashboard/data/top100-nfts.ts:1]: How about we take top 100 NFTs from Subgraph of OpenSea based on certain criteria. That would make it dynamic , freeing us from having to updates this list frequently

File: src/app/dashboard/page.tsx [hunk 1]
```diff
@@ -26,25 +25,56 @@ import useSWR from 'swr'
...
+import {
+  OsNft,
+  findAllNFTsOsApi,
+} from './utils/nft-balance-opensea'
+import { trimString } from './utils'
 
 const TABS = ['Tokens', 'Collectibles']
 
-const fetcher = async () => {
+const TestWalletAddress = '0xCeF4690B0976691E5e2ab92dd92A30D65E7733D8'
```
Comment (jainkrati) [inline src/app/dashboard/page.tsx:36]: pls add a todo here for removing in future

File: src/app/dashboard/utils/nft-balance-opensea.ts
Comment (jainkrati) [reply src/app/dashboard/utils/nft-balance-opensea.ts:1]: Lets cache the response in FE such that we refresh every 4 hours? Also, lets use multiple free accounts on a polling basis to further reduce requets per second

**Review by jainkrati — APPROVED** [review]

File: src/app/dashboard/data/top100-nfts.ts
[the remark this answers is not in the document]
Comment (mmahdigh) [reply src/app/dashboard/data/top100-nfts.ts:1]: Yes, that's a great idea. However, I couldn't find the endpoint for top NFT collections in Open Sea even though they do have such a page on their website. Perhaps, I need to look at the request being made from that page. (If it's client-side)

File: src/app/dashboard/utils/nft-balance-opensea.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import { axiosInstance } from '@/services/axiosInstance'
+import { SupportedChainIds, getChainIds } from '../data/supported_tokens'
+
+// Free personal API key
+const OpenSeaApiKey = 'd339363f19444a649ff0d29965f58b2e'
```
Comment (mateodaza) [inline src/app/dashboard/utils/nft-balance-opensea.ts:5]: Is this limited to our domains + localhost?

**Pull request merged**